src/gwt/src/org/rstudio/studio/client/workbench/views/files/Files.java

```diff
@@ -777,13 +838,75 @@ private void showFileInBrowser(FileSystemItem file)
       }
    }
 
+   // generate a filename based on the file type and currentPath_ variable
+   private FileSystemItem getDefaultFileName(TextFileType fileType) 
+   {
+      String defaultExt = fileType.getDefaultExtension();
+      
+      // extension has a '.' at the start, so remove that in the default name
+      String newFileDefaultName = "new_" + defaultExt.toUpperCase().substring(1) + "_file" + defaultExt.toLowerCase();
+      String path = workingPath_.completePath(newFileDefaultName);
+      FileSystemItem newTempFile = FileSystemItem.createFile(path);
+      return newTempFile;
+   }
+   
+   private void touchFile(TextFileType fileType)
+   {
+      // prepare default information about the new file
+      FileSystemItem newTempFile = getDefaultFileName(fileType);
+      
+      // guard for reentrancy
+      if (inputPending_)
+         return;
+      
+      inputPending_ = true;
+
+      // prompt for new file name then execute the operation
+      globalDisplay_.promptForText("Create and Open New File",
+                                   "Please enter the new file name:",
+                                   newTempFile.getName(),
+                                   0,
+                                   newTempFile.getStem().length(),
+                                   null,
+                                   new ProgressOperationWithInput<String>()
+      {
+         public void execute(String input, final ProgressIndicator progress)
+         {
+            // no longer waiting for user to input
+            inputPending_ = false;
+
+            progress.onProgress("Creating file...");
+
+            String path = currentPath_.completePath(input);
+            final FileSystemItem newFile = FileSystemItem.createFile(path);
+
+            // execute on the server
+            server_.touchFile(newFile, new VoidServerRequestCallback(progress)
```

Comment:
I think we should add an error handler here so we can show something better than a generic error message.

Reply:
On the RPC request or on the creation of the fileSystemItem itself?

Reply:
On the RPC request.

Reply:
I've overridden the default error display associated with the progress indicator with one that describes that the error occurred specifically in the file creation process.

This should help locate the error better, especially if the server returns a generic (or generic-looking) error string.

![image](https://user-images.githubusercontent.com/5544965/133133391-c34baf6e-40ee-4ae7-bc2d-c0173dc696f4.png)